packages/dev/codemods/src/s1-to-s2/src/codemods/changes.ts

```diff
@@ -1,1386 +0,0 @@
-  UpdatePlacementToSingleValueProps,
-  UpdatePropNameAndValueOptions,
-  UpdatePropNameOptions,
-  UpdatePropValueAndAddNewPropOptions,
-  UpdateToNewComponentOptions
-} from './transforms';
-
-type FunctionInfo =
-  | { name: 'commentOutProp', args: CommentOutPropOptions }
-  | { name: 'removeProp', args: RemovePropOptions }
-  | { name: 'updatePropNameAndValue', args: UpdatePropNameAndValueOptions }
-  | {
-      name: 'updatePropValueAndAddNewProp',
-      args: UpdatePropValueAndAddNewPropOptions
-    }
-  | { name: 'updatePropName', args: UpdatePropNameOptions }
-  | {
-      name: 'updateComponentIfPropPresent',
-      args: UpdateComponentIfPropPresentOptions
-    }
-  | { name: 'updateToNewComponent', args: UpdateToNewComponentOptions }
-  | { name: 'moveRenderPropsToChild', args: MoveRenderPropsOptions }
-  | { name: 'removeProp', args: RemovePropOptions }
-  | {
-      name: 'updateComponentWithinCollection',
-      args: UpdateComponentWithinCollectionOptions
-    }
-  | {
-      name: 'updateTabs',
-      args: {}
-    }
-  | {
-      name: 'movePropToNewChildComponent',
-      args: MovePropToNewChildComponentOptions
-    }
-  | {
-      name: 'movePropToParentComponent',
-      args: MovePropToParentComponentOptions
-    }
-  | {
-    name: 'convertDimensionValueToPx',
-    args: ConvertDimensionValueToPxOptions
-  }
-  | {
-    name: 'updatePlacementToSingleValue',
-    args: UpdatePlacementToSingleValueProps
-  }
-  | {
-    name: 'removeComponentIfWithinParent',
-    args: RemoveComponentIfWithinParentOptions
-  }
-  | {
-    name: 'addCommentToElement',
-    args: AddCommentToElementOptions
-  }
-  | {
-    name: 'commentIfParentCollectionNotDetected',
-    args: {}
-  }
-  | {
-    name: 'updateAvatarSize',
-    args: {}
-  }
-  | {
-    name: 'updateLegacyLink',
-    args: {}
-  }
-  | {
-    name: 'addColumnsPropToRow',
-    args: {}
-  }
-  | {
-    name: 'updateRowFunctionArg',
-    args: {}
-  }
-  | {
-    name: 'updateKeyToId',
-    args: {}
-  }
-  | {
-    name: 'updateDialogChild',
-    args: {}
-  }
-  | {
-    name: 'updateActionGroup',
-    args: {}
-  } | {
-    name: 'commentIfNestedColumns',
-    args: {}
-  } | {
-    name: 'addRowHeader',
-    args: {}
-  }
-
-type Change = {
-  description: string,
-  reason: string,
-  function: FunctionInfo
-};
-
-type ComponentChanges = {
-  changes: Change[]
-};
-
-type ChangesJSON = {
-  [component: string]: ComponentChanges
-};
-
-export const changes: ChangesJSON = {
-  Avatar: {
-    changes: [
-      {
-        description: 'Comment out isDisabled',
-        reason: 'It has not been implemented yet',
-        function: {
-          name: 'commentOutProp',
-          args: {propToComment: 'isDisabled'}
-        }
-      },
-      {
-        description: 'Update size prop',
-        reason: 'Updated naming convention',
-        function: {
-          name: 'updateAvatarSize',
-          args: {}
-        }
-      }
-    ]
-  },
-  ActionGroup: {
-    changes: [
```

Comment:
just checking, but the declarative approach is going away?

Reply:
Yep, I think there were just too many steps to remember to do when adding a codemod. Also with more singe component-specific codemods getting added it made less sense.